cloudflare/resource_cloudflare_notification_policy_webhooks.go

```diff
@@ -0,0 +1,175 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceCloudflareNotificationPolicyWebhooks() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareNotificationPolicyWebhooksCreate,
+		Read:   resourceCloudflareNotificationPolicyWebhooksRead,
+		Update: resourceCloudflareNotificationPolicyWebhooksUpdate,
+		Delete: resourceCloudflareNotificationPolicyWebhooksDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareNotificationPolicyWebhooksImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"secret": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_success": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_failure": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareNotificationPolicyWebhooksCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get("account_id").(string)
+
+	notificationWebhooks := buildNotificationPolicyWebhooks(d)
+
+	webhooksDestination, err := client.CreateNotificationWebhooks(context.Background(), accountID, &notificationWebhooks)
+
+	if err != nil {
+		return fmt.Errorf("error connecting webhooks destination %s: %s", notificationWebhooks.Name, err)
+	}
+	formattedWebhookID, err := uuid.Parse(webhooksDestination.Result.ID)
+	if err != nil {
+		return fmt.Errorf("error setting notification webhooks: %s", err)
+	}
+
+	d.SetId(formattedWebhookID.String())
+
+	return resourceCloudflareNotificationPolicyWebhooksRead(d, meta)
+}
+
+func resourceCloudflareNotificationPolicyWebhooksRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	webhooksDestinationID := d.Id()
+	accountID := d.Get("account_id").(string)
+
+	notificationWebhooks, err := client.GetNotificationWebhooks(context.Background(), accountID, webhooksDestinationID)
+
+	name := d.Get("name").(string)
+	if err != nil {
+		return fmt.Errorf("error retrieving notification webhooks %s: %s", name, err)
+	}
+	id, err := uuid.Parse(notificationWebhooks.Result.ID)
+	if err != nil {
+		return fmt.Errorf("error setting notification webhooks %s: %s", name, err)
+	}
+	d.Set("id", id.String())
```

Comment:
the `Create` should be suffice for setting the resource ID as we shouldn't need to change it after the fact. this value is often used as an "anchor" for lookups and should remain consistent.

side note: setting the resource ID should use `d.SetId()` as it is a special attribute.

Reply:
removed this now and verified that the tests are passing.